pkg/asset/agent/image/cache.go

```diff
@@ -266,5 +267,16 @@ func DownloadImageFile(baseURL string) (string, error) {
 	}
 	u.location = *parsedURL
 
-	return u.download(imageDataType)
+	for i := 0; i < 3; i++ {
+		path := ""
+		path, err = u.download(imageDataType)
+		if err == nil {
+			return path, nil
+		} else if i < 2 {
+			// Only log during first 2 tries.
+			logrus.Warnf("Error when obtaining RHCOS image file, retrying: %s", err)
```

Comment:
On a second thought, by reviewing this code, I think that using a retry.Do to wrap that portion of code:

```
resp, err := http.Get(u.location.String())
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	// Check server response
	if resp.StatusCode != http.StatusOK {
		return "", errors.Errorf("bad status: %s", resp.Status)
	}
```

Will allow to implement a more generic retry mechanism, will require fewer lines of code and it will also help in reducing the code duplication (instead of apply the retry outside of the `download` function). @bfournie wdyt?

Reply:
Done!